src/bomf/model/__init__.py

```diff
@@ -0,0 +1,54 @@
+"""
+general data models for migrations
+"""
+from typing import Any, Iterable, Optional, Protocol, Type, TypeVar
+
+import attrs
+import bo4e.bo.geschaeftsobjekt
+
+
+# pylint:disable=too-few-public-methods
+@attrs.define(kw_only=True, auto_attribs=True)
+class BusinessObjectRelation:
+    """
+    a business object relation describes the relation between two business object
+    """
+
+    relation_type: str = attrs.field(validator=attrs.validators.instance_of(str))
+    """
+    the relation type describes how two business objects relate to each other
+    """
+    relation_part_a: Any = attrs.field()
+    """
+    one Business Object or COM
+    """
+
+    relation_part_b: Any = attrs.field()
+    """
+    another Business Object or COM
+    """
+
+
+SpecificBusinessObject = TypeVar("SpecificBusinessObject", bound=bo4e.bo.geschaeftsobjekt.Geschaeftsobjekt)
+"""
+an arbitrary but fixed business object type
+"""
+
+
+class Bo4eDataSet(Protocol):
```

Comment:
Why does this class inherit from a type hint class?

Reply:
https://peps.python.org/pep-0544/
095a5907cd497dd96dc32ff9014628e8b4f4120c